src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2037,72 +2042,125 @@ namespace SymbolicSymbol {
             Vec<ASR::expr_t*>& args,
-            const std::function<void (const std::string &, const Location &)> err) {
-        if (args.size() != 2) {
-            err("Intrinsic Symbol Add operator accepts exactly 2 arguments", loc);
-        }
-
-        Vec<ASR::expr_t*> arg_values;
-        arg_values.reserve(al, args.size());
-        for( size_t i = 0; i < args.size(); i++ ) {
-            arg_values.push_back(al, ASRUtils::expr_value(args[i]));
-        }
-        ASR::expr_t* compile_time_value = eval_SymbolicAdd(al, loc, arg_values);
+            const std::function<void (const std::string &, const Location &)> /*err*/) {
+        ASR::expr_t* compile_time_value = eval_SymbolicPi(al, loc, args);
         ASR::ttype_t *to_type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
         return ASR::make_IntrinsicFunction_t(al, loc,
-                static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicAdd),
-                args.p, args.size(), 0, to_type, compile_time_value);
+                static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicPi),
+                nullptr, 0, 0, to_type, compile_time_value);
     }
 
-} // namespace SymbolicAdd
+} // namespace SymbolicPi
 
-namespace SymbolicPi {
+namespace SymbolicInteger {
 
     static inline void verify_args(const ASR::IntrinsicFunction_t& x, diag::Diagnostics& diagnostics) {
-        ASRUtils::require_impl(x.n_args == 0, "SymbolicPi does not take arguments",
+        ASRUtils::require_impl(x.n_args == 1,
+            "SymbolicInteger intrinsic must have exactly 1 input argument",
+            x.base.base.loc, diagnostics);
+
+        ASR::ttype_t* input_type = ASRUtils::expr_type(x.m_args[0]);
+        ASRUtils::require_impl(ASR::is_a<ASR::Integer_t>(*input_type),
+            "SymbolicInteger intrinsic expects an integer input argument",
             x.base.base.loc, diagnostics);
     }
 
-    static inline ASR::expr_t *eval_SymbolicPi(Allocator &/*al*/,
+    static inline ASR::expr_t* eval_SymbolicInteger(Allocator &/*al*/,
     const Location &/*loc*/, Vec<ASR::expr_t*>& /*args*/) {
         // TODO
         return nullptr;
     }
 
-    static inline ASR::asr_t* create_SymbolicPi(Allocator& al, const Location& loc,
+    static inline ASR::asr_t* create_SymbolicInteger(Allocator& al, const Location& loc,
             Vec<ASR::expr_t*>& args,
             const std::function<void (const std::string &, const Location &)> /*err*/) {
-        ASR::expr_t* compile_time_value = eval_SymbolicPi(al, loc, args);
+        // if (args.size() != 1) {
+        //     err("Intrinsic SymbolicInteger function accepts exactly 1 argument", loc);
+        // }
+
+        // ASR::ttype_t* type = ASRUtils::expr_type(args[0]);
+        // if (!ASRUtils::is_integer(*type)) {
+        //     err("Argument of the SymbolicInteger function must be an Integer",
+        //         args[0]->base.loc);
+        // }
```

Comment:
I've commented these right now because I realized that essentially we won't be encountering this case anyways through the Casting operation . The call for `create_SymbolicInteger` is made in `make_Cast_t_value` in `asr_utils.cpp`
and as per how the function is defined , we are only passing the single argument which is being casted (`a_arg`)
```
ASR::asr_t* make_Cast_t_value(Allocator &al, const Location &a_loc,
            ASR::expr_t* a_arg, ASR::cast_kindType a_kind, ASR::ttype_t* a_type) {
```

So we won't encountering the `if (args.size() != 1) ` case in `create_SymbolicInteger` anyways , so these error statements become redundant. Maybe I could take these out !